### PR TITLE
Refactor is_target_maintainer

### PR DESCRIPTION
**Pull request by coolo**
The old code was a little hard to parse

### CODE REVIEW

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -927,26 +927,8 @@ def auto_accept
...
-      end
-      tprj = Project.find_by_name(a.target_project)
-      if tprj
-        has_target = true
-        is_target_maintainer &= user.can_modify?(tprj)
-      end
-    end
-    has_target && is_target_maintainer
+  def is_target_maintainer?(user)
+    bs_request_actions.map { |action| action.is_target_maintainer?(user) }.all?
```
Comment (bgeuken) [inline src/api/app/models/bs_request.rb]: wennschon dennschon:smiley:
```ruby
 bs_request_actions.all? { |action| action.is_target_maintainer?(user) }
```

Comment (coolo) [reply src/api/app/models/bs_request.rb]: Hmm, yesterday I tried and was somehow led to believe that .all? would return false on empty array. Today it does not - which is a problem

Comment (coolo) [reply src/api/app/models/bs_request.rb]: well, it's not - as we have no requests without actions. just the old `has_target` logic looked for it. But an action would target will return false on is_target_action? (I guess I could write specs for that)

Comment (coolo) [reply src/api/app/models/bs_request.rb]: done that now - sorry for the early morning brabling :)

File: src/api/spec/models/bs_request_action_spec.rb [hunk 2]
```diff
@@ -143,5 +143,36 @@
...
+
+      context 'on home target' do
+        let(:another_user) { create(:confirmed_user) }
+        let(:bs_request) do
+          create(:set_bugowner_request, target_project: user.home_project)
+        end
+        let(:bs_request_action) { bs_request.bs_request_actions.first }
+
+        it 'is true for user' do
+          expect(bs_request_action).to be_is_target_maintainer(user)
```
Comment (coolo) [inline src/api/spec/models/bs_request_action_spec.rb]: that looks very strange, but that's how rubocop wrote it
